pkg/syncer/shared/namespace.go

```diff
@@ -49,11 +51,18 @@ func LocatorFromAnnotations(annotations map[string]string) (*NamespaceLocator, e
 
 // PhysicalClusterNamespaceName encodes the NamespaceLocator to a new
 // namespace name for use on a physical cluster. The encoding is repeatable.
-func PhysicalClusterNamespaceName(l NamespaceLocator) (string, error) {
+func PhysicalClusterNamespaceName(l NamespaceLocator, workloadClusterName string) (string, error) {
```

Comment:
is this WorkloadCluster name or workload ClusterName?

Reply:
WorkloadCluster Name

Reply:
would drop the "Name" suffix to make it clear.

Note that the name of the workload cluster is not unique in kcp. We need more here.

Reply:
I was under the impression that NamespaceLocator(Workspace + Namespace) + WorkloadCluster should be a unique identifier... there's some case I'm missing?

Reply:
yes and no. The WorkloadCluster lives in a different location workspace. Per `NamespaceLocator(Workspace, ..)` there is only one such location workspace, but on a pcluster this does not help as multiple syncer can point to different location workspaces with the same WorkloadCluster names.